### PR TITLE
Upgrade realm dependency for Xcode compatibility

### PR DESCRIPTION
**Pull request by tedlee**
Xcode 9 made some changes to how c++ files are handled which broke our Realm 2.1.0 dependency. Upgrading to the latest Realm build, 2.10.1, fixes the issue. 

@21echoes - we'll use this in testing for a while before merging to ensure there are no unforeseen consequences

### CODE REVIEW

File: Realm-JSONAPI/Classes/RLMObject+JSONAPISerializer.m [hunk 1]
```diff
@@ -29,11 +29,7 @@ - (NSMutableDictionary *)toJSONWithIncludedTypes:(NSSet *)types
   NSDictionary        *map = [[self class] JSONtoModelMap];
   RLMSchema           *schema = self.realm ? self.realm.schema : [RLMRealm defaultRealm].schema;
   Class                class = [self class];
-  // Realm often subclasses our objects with RLMAccessor, or RLMStandalone if we're not in a Realm
-  if ([NSStringFromClass(class) rangeOfString:@"RLMAccessor"].location != NSNotFound
-      || [NSStringFromClass(class) rangeOfString:@"RLMStandalone"].location != NSNotFound) {
-    class = [class superclass];
-  }
```
Comment (21echoes) [inline Realm-JSONAPI/Classes/RLMObject+JSONAPISerializer.m:36]: any reason that these code blocks were removed? did Realm 2.10 fix the need for this?

**Comment by 21echoes** [comment]
can we also see if we can fix the CircleCI tests before we merge this?

File: Realm-JSONAPI/Classes/RLMObject+JSONAPISerializer.m (hunk 1 above)
Comment (tedlee) [reply Realm-JSONAPI/Classes/RLMObject+JSONAPISerializer.m:36]: In testing with Realm 2.10.1 `[self class]` was of type RLM:Managed (previously RLM:Accessor) which resulted in never grabbing the correct superclass

**Comment by phildini** [comment]
Closing old PRs, feel free to re-open if still needed.

**Pull request closed without merging**